src/main/java/reposense/util/FileUtil.java

```diff
@@ -122,8 +127,70 @@ public static void copyDirectoryFiles(Path src, Path dest) throws IOException {
         }
     }
 
+    /**
+     * Generates a list of Path contained in the given {@code directoryName} directory.
+     */
+    public static List<Path> listJsonFiles(String directoryName) {
+        Path directory = Paths.get(directoryName);
+        List<Path> allJsonFiles = null;
+        try {
+            allJsonFiles = Files.walk(directory)
+                    .filter(p -> p.toString().endsWith(".json"))
+                    .distinct()
+                    .collect(Collectors.toList());
+        } catch (IOException ioe) {
+            logger.log(Level.SEVERE, ioe.getMessage(), ioe);
+        }
+
+        return allJsonFiles;
+    }
+
+    /**
+     * Zips all the JSON files contained in the {@generatedFolder} directory.
+     * Creates the zip folder in the {@code directoryPath}
+     */
+    public static void zip(String directoryPath, String generatedFolder) {
+
+        String zipFileName = "archiveJSON.zip";
+
+        List<Path> allJsonFiles = listJsonFiles(directoryPath + File.separator + generatedFolder);
+
+        //byte buffer for I/O
+        ByteBuffer buffer = ByteBuffer.allocate(1 << 10);
+        int length;
+        try (FileOutputStream fos = new FileOutputStream(directoryPath + File.separator + zipFileName);
+             ZipOutputStream zos = new ZipOutputStream(fos)) {
+            for (int i = 0; i < allJsonFiles.size(); i++) {
+                try (FileInputStream fis = new FileInputStream(allJsonFiles.get(i).toFile())) {
+
+                    // begin writing a new ZIP entry, positions the stream to the start of the entry data
+                    if (isSummaryJson(allJsonFiles.get(i))) {
```

Comment:
Same problem with the hard-coded ```archiveJSON.zip```. This method is very specific to be in ```FileUtil```.

Perhaps, you can generalize main zipping logic of this method, then use another wrapper to do a specific function.

Can discuss with @yamidark. If you need a code example, just @mention me.